File: app/src/main/java/com/lkms/data/model/java/Experiment.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "Experiment".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Experiment {

    @SerializedName("experimentId")
    private Integer experimentId;

    @SerializedName("experimentTitle")
    private String experimentTitle;

    @SerializedName("objective")
    private String objective;

    @SerializedName("experimentStatus")
    private String experimentStatus;

    // Kiểu 'date' trong SQL thường được xử lý như String trong client
    @SerializedName("startDate")
    private String startDate;

    @SerializedName("finishDate")
    private String finishDate;

    // Khóa ngoại đến User
    @SerializedName("userId")
    private Integer userId;

    // Khóa ngoại đến Protocol
    @SerializedName("protocolId")
    private Integer protocolId;

    // Khóa ngoại đến Project
    @SerializedName("projectId")
    private Integer projectId;
}

File: app/src/main/java/com/lkms/data/model/Experiment.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Experiment(
    @SerialName("experimentId")
    val experimentId: Int? = null,

    @SerialName("experimentTitle")
    val experimentTitle: String? = null,

    @SerialName("objective")
    val objective: String? = null,

    @SerialName("experimentStatus")
    val experimentStatus: String? = null,

    // Kiểu 'date' trong SQL thường được xử lý như String trong client
    @SerialName("startDate")
    val startDate: String? = null,

    @SerialName("finishDate")
    val finishDate: String? = null,

    // Khóa ngoại đến User
    @SerialName("userId")
    val userId: Int? = null,

    // Khóa ngoại đến Protocol
    @SerialName("protocolId")
    val protocolId: Int? = null,

    // Khóa ngoại đến Project
    @SerialName("projectId")
    val projectId: Int? = null
)